QuantConnect.TDAmeritradeBrokerage/TDCliCredentialProvider.cs

```diff
@@ -0,0 +1,63 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TDAmeritradeApi.Client;
+
+namespace QuantConnect.Brokerages.TDAmeritrade
+{
+    /// <summary>
+    /// TD Ameritrade API credential provider for inputting information by the command line.
+    /// </summary>
+    public class TDCliCredentialProvider : ICredentials
```

Comment:
The CliCredentialProvider was really only ever for use on the desktop. My intent was that the website would interact with the user when they sign up to use the TDA brokerage to gather the required information. For the most part it only needs to be done once. Unless TDA has some issue and needs you to re-auth. I think it's only happened to me once.

Reply:
The refresh token is good for a month and a half so as long as the api is used within that time period there will be no issues. I don't see it being a problem with someone who is live trading because the api is being used all the time.